### ISSUE TITLE
add support for Lombok annoatations Showing 'Missing Ctor ' 

### ISSUE THREAD

**Issue by mariyacherian** [checkstyle/checkstyle#13519]
I have checked https://github.com/checkstyle/checkstyle/issues/2579
am facing similar problem like this.When am giving @NoArgsConstructor to a service class this shows the 'Missing Ctor ' rule .scrrenshot attached. found that lombok genereated the constructor for the class by decoding the class files. here is the sample code
```bash
@NoArgsConstructor
@Service
public class ReceiveMessageJsonHandler

{
	private static final String SAND = "sand";
 }
```

```bash
I have gradle(6.9) ,open jdk 13  ,springboot(2.7.0) project with following dependecy

compileOnly 'org.projectlombok:lombok:1.18.22'
annotationProcessor 'org.projectlombok:lombok:1.18.22'
testCompileOnly 'org.projectlombok:lombok:1.18.22'
testAnnotationProcessor 'org.projectlombok:lombok:1.18.22'
```

---

**am not intereseted to skip this rule because there is no violation of the rule actually the constructor is generated the issue due to the avoiding of @generated code in the class file.
![Screenshot from 2023-08-07 11-23-46](https://github.com/checkstyle/checkstyle/assets/85922500/545a5d5c-27e3-4922-9f91-119509247451)
**

---
it's better to support lombok generated codes

**Comment by romani** [comment]
Please use https://checkstyle.org/filters/suppressionxpathsinglefilter.html
To suppress non desirable violations.

We can not support all tools that does code generation during compilation time.
